### DISCUSSION TITLE
Themes: making transitions easy.

### DISCUSSION THREAD

**Discussion by paulovbettio** [godotengine/godot-proposals#4438] [GUI]
Themes should have a built-in method to tween values between StyleBoxes using a custom Curve, like CSS _transition_ tag.

For example: we have a Button. This button have three StyleBoxFlat for its states, red when normal, pink when hovered and white when pressed. If we enable the transition on this Button class, the StyleBoxes will smoothly transition it's values to the values of the StyleBox of next state, so red will transition to pink when hovered, instead of just swapping states without animation.

**Comment by Calinou** [comment]
Morphing all of a StyleBoxFlat's properties over time is likely possible, but it will require making the StyleBoxFlat instance unique on a per-button basis.

**Reply by paulovbettio** [reply]
One approach could be using an instantiated style box for the transition and then replacing it back when the transition ends.

**Comment by acgc99** [comment]
What I do for this is taking a screenshot and animating (`Tween`) `modulate.a` of a `TextureRect` holding that screenshot. It gives nice results.

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/8992, which is more detailed.
